java/client/src/main/java/glide/api/models/commands/SetOptions.java

```diff
@@ -0,0 +1,171 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import static glide.api.models.commands.SetOptions.ExpiryType.KEEP_EXISTING;
+import static glide.api.models.commands.SetOptions.ExpiryType.MILLISECONDS;
+import static glide.api.models.commands.SetOptions.ExpiryType.SECONDS;
+import static glide.api.models.commands.SetOptions.ExpiryType.UNIX_MILLISECONDS;
+import static glide.api.models.commands.SetOptions.ExpiryType.UNIX_SECONDS;
+
+import glide.api.commands.StringCommands;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import redis_request.RedisRequestOuterClass.Command;
+
+/**
+ * Optional arguments for {@link StringCommands#set(String, String, SetOptions)} command.
+ *
+ * @see <a href="https://redis.io/commands/set/">redis.io</a>
+ */
+@Builder
+public final class SetOptions {
+
+    /**
+     * If <code>conditionalSet</code> is not set the value will be set regardless of prior value
+     * existence. If value isn't set because of the condition, command will return <code>null</code>.
+     */
+    private final ConditionalSet conditionalSet;
```

Comment:
this should be an optional, in order to properly represent that this value can be not set.
It's better to make types explicit than rely on enum being null, which is a (weird) language specific ability.